runtime/compilers/rillv1/parse_model.go

```diff
@@ -106,6 +112,44 @@ func (p *Parser) parseModel(node *Node) error {
 		inputProps["sql"] = sql
 	}
 
+	// special handling to mark model as updated when local file changes
+	if inputConnector == "local_file" {
+		c, ok := inputProps["invalidate_on_change"].(bool)
+		invalidateModelOnUpdate := ok && c
+		path, ok := inputProps["path"].(string)
+		if ok && invalidateModelOnUpdate {
+			entries, err := p.Repo.ListRecursive(ctx, path, true)
+			if err == nil && len(entries) > 0 {
+				var localPaths []string
+				root, err := p.Repo.Root(ctx)
+				if err != nil {
+					return err
+				}
+
+				// Update parser's localDataToResourcepath map to track which resources depend on the local file
+				for _, entry := range entries {
+					localPaths = append(localPaths, filepath.Join(root, entry.Path))
```

Comment:
The same parser is only used for one repo, so could it just store the project-relative path? That would reduce dependency on the local filesystem.

I suppose there's a case on local where you can reference a path outside of the project. I would actually prefer we do *not* hash or auto-refresh for files outside the project because a) it's a workflow we should discourage, and b) the watcher doesn't (and shouldn't) watch them.